### PR TITLE
Edited Game of Life in Python

### PR DESCRIPTION
**Pull request by 2Clutch**
Congrats on taking the first step to contributing to the [Sample Programs in Every Language](https://therenegadecoder.com/code/sample-programs-in-every-language/) repository!

For simplicity, please make sure that your pull request includes one and only one sample program and language.

## Please Complete the Following

- [x] I would **NOT** like credit for my contribution in the article (credit will be included otherwise)
- [ ] I plan to write the article for this code snippet
- [x] I named the pull request using `<Sample Program> in <Language>` format
- [x] I created/updated the language README
  - [x] I added the sample program name to the README
  - [x] I added fun facts (i.e. debut, developer, typing, etc.)
  - [x] I added reference link(s) to the README
- [ ] I fixed #your-issue-number-here

## Notes

Just a little background, I knew little about "The Game of Life" before working on this. I'm sure not sure I understand it, but that's beyond the point.

I tried to implement it using canvas as requested per the issue. However while my code looks fine to me, the GUI is not responding for whatever reason. I added a `start` and `stop` feature for convenience. 

When the program is ran from the command line, the GUI loads up as expected with no issue. When I press `start` however, the program crashes and I have to exit my terminal prompt to bring things back to normal.

If you could please take a look and provide some feedback, I'd sincerely appreciate it. I'm stuck and I am not sure what to do next.

### CODE REVIEW

**Comment by jrg94** [comment]
Awesome! I'm excited to see you've tackled #110. I'll take a look, but I'm really not well versed in tk at all. I find the library a little clunky.

File: archive/p/python/game-of-life.py [hunk 1]
```diff
@@ -1,108 +1,120 @@
...
+
+def click_colour_change(event):
+	print (event.x, event.y)
+	x, y = rectangle_coordinates(event.x, event.y)
+	try:
+		iy = x / 10 - 1
+		ix = y / 10 - 1
+		if ix == -1 or iy == -1:
+			raise IndexError
+			if grid[ix][iy].isAlive:
```
Comment (jrg94) [inline archive/p/python/game-of-life.py]: I opened up the code in PyCharm quickly, and I noticed there's an error here. These if statements are unreachable due to the raising of the IndexError.

File: archive/p/python/game-of-life.py [hunk 2]
```diff
@@ -1,108 +1,120 @@
...
+def status_update(cell):
+	num_alive = 0
+	x, y = cell.pos_matrix
+	for i in (x-1, x, x+1):
+		for j in (y-1, y, y+1):
+			if i == x and j == y:
+				continue
+				if i == -1 or j == -1:
+					continue
+					try:
```
Comment (jrg94) [inline archive/p/python/game-of-life.py]: Here's another section of unreachable code. I believe the continue will jump to the next iteration of the loop, but don't quote me on that. 😆

File: archive/p/python/game-of-life.py [hunk 3]
```diff
@@ -1,108 +1,120 @@
...
+				j.nextStatus = not j.isAlive
+				print ("change in", j.pos_matrix, "from", j.isAlive, "to", j.nextStatus)
+			else:
+				j.nextStatus = j.isAlive
+				paint_grid()
+				global game_id
+				game_id = root.after(200, start_game)
+
+def end_game():
+	root.after_cancel(begin_id)
```
Comment (jrg94) [inline archive/p/python/game-of-life.py]: PyCharm claims `begin_id` doesn't exist. I didn't dig into this too much.

**Comment by 2Clutch** [comment]
Just went and fixed the `begin_id` issue. I had called it `begin_id` that in the beginning, but renamed it `game_id` and forgot to update it there. I use Sublime Text as an editor. I will look around for a `PythonLinter` plugin of sort and trying to work things through.

**Comment by jrg94** [comment]
Sounds good! I'll keep playing a bit. I see the grid shows up, and I noticed your debugs are pretty helpful.

**Comment by jrg94** [comment]
We haven't touched this for a few months, so I'm going to close this for now.

**Comment by jrg94** [comment]
And, we're back! 💃

**Comment by 2Clutch** [comment]
Does everything look ok?

**Comment by jrg94** [comment]
Just now getting a chance to look this over.

File: archive/p/python/palindrome.py [hunk 4]
```diff
@@ -0,0 +1,8 @@
+string = str(input("Enter the string:"))
```
Comment (jrg94) [inline archive/p/python/palindrome.py]: Thanks for moving this file! I didn't realize until now that it was sitting in root.

**Pull request merged**